flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/sink/BaseDeltaTaskWriter.java

```diff
@@ -74,11 +76,23 @@ public void write(RowData row) throws IOException {
       case INSERT:
       case UPDATE_AFTER:
         if (upsert) {
-          writer.delete(row);
+          // Upserts come in modeled as INSERT. We need to be sure that
```

Comment:
It's best to avoid personal pronouns in comments and docs. It isn't clear who "we" is and omitting it is typically much more direct. For example: "The columns that are not equality ID columns may have changed in the inserted row, so the delete file must be written using only the equality ID columns."